src/middlewares/authenticateToken.js

```diff
@@ -19,11 +20,26 @@ export function authenticateToken(req, res, next) {
     return next(Boom.unauthorized('Token not provided'));
   }
 
-  jwt.verify(token, ACCESS_TOKEN_SECRET, (err, user) => {
+  jwt.verify(token, ACCESS_TOKEN_SECRET, async (err, tokenUser) => {
     if (err) {
       return next(Boom.unauthorized('Invalid or expired token'));
     }
-    req.user = user;
+
+    const user = await User.where({ id: tokenUser.id }).fetch({ withRelated: ['roles'] });
+
+    if (!user) {
+      return next(Boom.unauthorized('User not found'));
+    }
+
+    const userData = user.filterSensitiveData();
+
+    // Include roles in the response
+    const roles = user.related('roles').toJSON();
+
+    userData.roles = roles;
+
+    req.user = userData;
```

Comment:
```suggestion
	  const userData = user.filterSensitiveData();
	
	  // Attach roles to userData
	  req.user = {
	    ...userData,
	    roles: user.related('roles').toJSON(),
	  };
```
Also, filterSensitiveData already serializes to JSON. can we have single toJSON to serialize at once?